src/schema/date_time_options.rs

```diff
@@ -0,0 +1,606 @@
+use std::collections::{HashSet, VecDeque};
+use std::ops::BitOr;
+use std::sync::{Arc, Mutex};
+
+use serde::{Deserialize, Serialize};
+use time::format_description::well_known::{Iso8601, Rfc2822, Rfc3339};
+use time::OffsetDateTime;
+
+use super::Cardinality;
+use crate::schema::flags::{FastFlag, IndexedFlag, SchemaFlagList, StoredFlag};
+use crate::DateTimePrecision;
+
+/// Defines how DateTime field should be handled by tantivy.
+#[derive(Clone, Serialize, Deserialize, Derivative)]
+#[derivative(Debug, Eq, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct DateTimeOptions {
+    indexed: bool,
+    // This boolean has no effect if the field is not marked as indexed true.
+    fieldnorms: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    fast: Option<Cardinality>,
+    stored: bool,
+
+    // Accepted input format for parsing input as DateTime.
+    #[serde(default = "default_input_formats")]
+    input_formats: HashSet<DateTimeFormat>,
+
+    // Internal storage precision, used to avoid storing
+    // very large numbers when not needed. This optimizes compression.
```

Comment:
Does it apply to 
- indexing
- stored
- fast 
- index sorting
or only
- fast
or only 
-fast
-indexing?

It is not an obvious question and it should be detailed here.